homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,232 @@
+                self._name = name
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Failed to get name")
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return 16
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return 30
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._supported_features
+
+    @property
+    def name(self):
+        """Return the name of the aircon."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._aircon.said
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._aircon.get_online()
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._aircon.get_current_temp()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._aircon.get_temp()
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        await self._aircon.set_temp(kwargs.get(ATTR_TEMPERATURE))
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._aircon.get_current_humidity()
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        return self._aircon.get_humidity()
+
+    @property
+    def target_humidity_step(self):
+        """Return the supported step of target humidity."""
+        return 10
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        await self._aircon.set_humidity(humidity)
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return [HVAC_MODE_COOL, HVAC_MODE_HEAT, HVAC_MODE_FAN_ONLY, HVAC_MODE_OFF]
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, fan."""
+        if not self._aircon.get_power_on():
+            return HVAC_MODE_OFF
+
+        mode: AirconMode = self._aircon.get_mode()
+        return AIRCON_MODE_MAP.get(mode, None)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self._aircon.set_power_on(False)
+            return
+
+        mode = None
+        for k, val in AIRCON_MODE_MAP.items():
```

Comment:
Create an inverse dict here and then do `mode = HVAC_MODE_TO_AIRCON_MODE.get(hvac_mode)`

Reply:
ok